src/AzureAppConfigurationImpl.ts

```diff
@@ -60,25 +101,111 @@ export class AzureAppConfigurationImpl extends Map<string, unknown> implements A
             };
             if (this.#requestTracingEnabled) {
                 listOptions.requestOptions = {
-                    customHeaders: this.#customHeaders()
+                    customHeaders: this.#customHeaders(requestType)
                 }
             }
 
             const settings = this.#client.listConfigurationSettings(listOptions);
 
             for await (const setting of settings) {
                 if (setting.key) {
-                    const [key, value] = await this.#processAdapters(setting);
-                    const trimmedKey = this.#keyWithPrefixesTrimmed(key);
-                    keyValues.push([trimmedKey, value]);
+                    const keyValuePair = await this.#processKeyValues(setting);
+                    keyValues.push(keyValuePair);
+                }
+                // update etag of sentinels if refresh is enabled
+                if (this.#refreshEnabled) {

```

Comment:
If a watched setting is not loaded as part of the kv selectors, how do we update its etag in `this.sentinels`?

Reply:
> If a watched setting is not loaded as part of the kv selectors

Good question. Is that a valid case? To have a dedicated watched setting but not loaded from any selector? I doubt that a little bit from a user's perspective. 

> how do we update its etag in this.sentinels?

Back to the question. For the change-detected sentinel, its etag is updated immediately. But if there are multiple sentinels and since we break the loop and skip sending requests to check rest sentinels, their etags are not updated at that time. They are expected to be updated during the new load here. The consequence in such case is, if I update 2 sentinels at the same time, it might eventually refresh twice, although eventually the etags will be updated.

Reply:
My initial thought is as below, and we don't mess them up. It's much clearer.
- `settingSelector` specifies what to load
- `watchedSetting` specifies sentinels
Synced with Zhenlan offline, learning that dotnet provider used to be this behavior, but per customer feedback/request, watched settings are also loaded even not specified as part of selectors. 

I'll change to have consistent behavior with dotnet provider, loading watched settings even when they are not part of selectors. In terms of precedence, watched settings will have higher precedence.

Reply:
implemented in 5d80ccc